data/remote/src/main/java/com/brunotiba/remote/api/LocationService.kt

```diff
@@ -0,0 +1,38 @@
+package com.brunotiba.remote.api
+
+import com.brunotiba.remote.BuildConfig
+import com.brunotiba.remote.network.QueryParameter
+import com.brunotiba.remote.provider.RetrofitProvider
+import toothpick.InjectConstructor
+
+/**
+ * Service responsible for fetching data from the geolocation api.
+ */
+@InjectConstructor
+internal class LocationService(retrofitProvider: RetrofitProvider) : ApiService(retrofitProvider) {
+
+    override val apiUrl: String
+        get() = BuildConfig.GEOLOCATION_API_URL
+
+    override fun getMandatoryParams(): List<QueryParameter> =
+        listOf(
+            QueryParameter(API_KEY_QUERY, BuildConfig.GEOLOCATION_API_KEY),
+            QueryParameter(API_FORMAT_QUERY, API_FORMAT_VALUE)
+        )
+
+    private val locationApi: LocationApi = getApi()
```

Comment:
Suggestion: Wouldn't be better to have the result from getApi() stored in a lazy parameter inside the parent class? This way you can avoid having this private val in every implementation of ApiService and ensures that this snippet will be executed only once.

Reply:
This change would probably require a large effort for the obtained benefit.